recipes/ClockReactHooks/src/Main.purs

```diff
@@ -0,0 +1,89 @@
+module ClockReactHooks.Main where
+
+import Prelude
+import Data.JSDate (JSDate)
+import Data.JSDate as JSDate
+import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Effect (Effect)
+import Effect.Exception (throw)
+import Effect.Timer (clearInterval, setInterval)
+import Math (cos, pi, sin)
+import React.Basic.DOM (render)
+import React.Basic.DOM.SVG as SVG
+import React.Basic.Hooks (Component, Hook, JSX, UseEffect, UseState, coerceHook, component, useEffectOnce, useState', (/\))
+import React.Basic.Hooks as React
+import Web.DOM.NonElementParentNode (getElementById)
+import Web.HTML (window)
+import Web.HTML.HTMLDocument (toNonElementParentNode)
+import Web.HTML.Window (document)
+
+type Time
+  = { hours :: Number, minutes :: Number, seconds :: Number }
+
+main :: Effect Unit
+main = do
+  container <- getElementById "root" =<< map toNonElementParentNode (document =<< window)
+  case container of
+    Nothing -> throw "Root element not found."
+    Just c -> do
+      clock <- mkClock
+      render (clock {}) c
+
+mkClock :: Component {}
+mkClock = do
+  now <- JSDate.now >>= getTime
+  component "Clock" \_ -> React.do
+    { hours, minutes, seconds } <- useCurrentTime now
+    pure
+      $ SVG.svg
+          { viewBox: "0 0 400 400"
+          , width: "400"
+          , height: "400"
+          , children:
+              [ SVG.circle { cx: "200", cy: "200", r: "120", fill: "#1293D8" }
+              , hand 6 60.0 (hours / 12.0)
+              , hand 6 90.0 (minutes / 60.0)
+              , hand 3 90.0 (seconds / 60.0)
+              ]
+          }
+
+hand :: Int -> Number -> Number -> JSX
+hand width length turns =
+  let
+    t = 2.0 * pi * (turns - 0.25)
```

Comment:
```suggestion
    t = tau * (turns - 0.25)
```
https://pursuit.purescript.org/packages/purescript-math/docs/Math#v:tau

Reply:
Good call! I had blindly copied that from the Elm example verbatim, but looks like Elm just has `pi` and not `tau` in its standard library.